src/utils/utils.ts

```diff
@@ -177,3 +177,17 @@ export const getParsedData = (toolCall: any) => {
     return { ok: false, data: null };
   }
 };
+
+export const isGraphValidType = (graph: ICODAPGraph) => {
+  const {
+    plotType,
+    topSplitAttributeID: topId,
+    y2AttributeID: y2Id,
+    xAttributeID: xId,
+    yAttributeID: yId,
+    rightSplitAttributeID: rightId
+  } = graph;
+  const isDotPlot = plotType === "dotPlot" || plotType === "binnedDotPlot";
+  const isUnivariateDotPlot = isDotPlot && !topId && !rightId && !y2Id && ((xId && !yId) || (yId && !xId));
+  return graph.plotType === "scatterPlot" || isUnivariateDotPlot;
+};
```

Comment:
There's a good amount of shared code here with the `isUnivariateDotPlot`, `isScatterPlot`, and `isValidType` views in `CODAPGraphModel`. Could you maybe define `isUnivariateDotPlot` and `isScatterPlot' as utility functions in `utils.ts`, refactor `isGraphValidType` to call those, and then update the views on the `CODAPGraphModel` to each call their related utility function?